core/src/main/scala/org/scalajs/macrotaskexecutor/MacrotaskExecutor.scala

```diff
@@ -67,17 +66,15 @@ object MacrotaskExecutor extends ExecutionContextExecutor {
         if (currentlyRunningATask) {
           js.Dynamic.global.setTimeout(() => runIfPresent(handle), 0)
         } else {
-          tasksByHandle.get(handle) match {
-            case Some(task) =>
-              currentlyRunningATask = true
-              try {
-                task()
-              } finally {
-                tasksByHandle -= handle
-                currentlyRunningATask = false
-              }
-
-            case None =>
+          val task = tasksByHandle(handle).asInstanceOf[js.UndefOr[Runnable]]
+          if (task.isDefined) {
+            currentlyRunningATask = true
+            try {
+              task.asInstanceOf[Runnable].run()
```

Comment:
If you're going to cast here anyway, using `js.UndefOr` only to get `isDefined` is more trouble than it's worth. You can use `js.isUndefined(task)` instead.